jwst/outlier_detection/outlier_detection_scaled_step.py

```diff
@@ -31,7 +31,7 @@ class OutlierDetectionScaledStep(Step):
         scale = string(default='0.5 0.4')
         backg = float(default=0.0)
         save_intermediate_results = boolean(default=False)
-        good_bits = integer(default=6)
+        good_bits = string(default="~DO_NOT_USE")  # DQ flags to allow
```

Comment:
So this presents a real conundrum. If the `outlier_detection` step itself sets both DO_NOT_USE and OUTLIER for pixels that it finds to be outliers, then if a user runs the step again to iterate on something like the rejection threshold, the pixels previously flagged with DO_NOT_USE won't get used in subsequent runs. And we can't implement logic like if a pixel has both DO_NOT_USE and OUTLIER set, assume it's just from a previous run of the step and go ahead and use the pixel, because the DO_NOT_USE may have been set for some other legit reason because the pixel really is bad. In which case, what's the point of having this param at all? Maybe we should consider just giving up on it and if folks want to iterate the step, they'll just have to reset the DQ flags manually before rerunning.

Oh, but wait!  We're not updating the input images in place, are we? The updated DQ flag values only appear in the *output* images that are created. So a user really could just go ahead and rerun the step on the original input images. This is what's different compared to the HST pipeline usage: those updated the inputs in place.

So maybe this default setting (use any pixel that's NOT flagged with DO_NOT_USE) is OK after all and actually achieves what we want!

Never mind ...

Reply:
Actually, we don't even need `good_bits` at all.  We are not modifying images in place.  We are always starting with input `_cal` files and generating `_crf` files.  I think `good_bits` is just confusing and not a desirable workflow.

Reply:
Actually, we don't even need `good_bits` at all.  We are not modifying images in place.  We are always starting with input `_cal` files and generating `_crf` files.  I think `good_bits` is just confusing and not a desirable workflow.

Reply:
I suspect this was implemented for HST because all the instruments flag their DQ bits differently.  This is not the case with JWST.

Reply:
@hbushouse Just to clarify before merging, with regard to this thread, the `good_bits` parameter will be left in place?

Reply:
It can't hurt to leave it in, at least for now. In the default mode it's at least doing the right thing now, in terms of rejecting pixels that have DO_NOT_USE in the input and also setting DO_NOT_USE in the output (for use downstream). It's likely that we don't really need it, because we aren't modifying the inputs in place, but it can't hurt to have the flexibility available (it's amazing what use cases folks will come up with that we never thought of).